File: TransCoder_tokenized_test_set_functions/cpp/FIND_KTH_CHARACTER_OF_DECRYPTED_STRING.cpp
char encodedChar ( string str, int k ) {
  string expand = "";
  string temp;
  int freq = 0;
  for ( int i = 0;
  str [ i ] != '\0';
  ) {
    temp = "";
    freq = 0;
    while ( str [ i ] >= 'a' && str [ i ] <= 'z' ) {
      temp . push_back ( str [ i ] );
      i ++;
    }
    while ( str [ i ] >= '1' && str [ i ] <= '9' ) {
      freq = freq * 10 + str [ i ] - '0';
      i ++;
    }
    for ( int j = 1;
    j <= freq;
    j ++ ) expand . append ( temp );
  }
  if ( freq == 0 ) expand . append ( temp );
  return expand [ k - 1 ];
}

File: TransCoder_tokenized_test_set_functions/python/FIND_KTH_CHARACTER_OF_DECRYPTED_STRING.py
def encodedChar ( str, k ) :
    expand = ""
    freq = 0
    i = 0
    while ( i < len ( str ) ) :
        temp = ""
        freq = 0
        while ( i < len ( str ) and ord ( str [ i ] ) >= ord ( 'a' ) and ord ( str [ i ] ) <= ord ( 'z' ) ) :
            temp += str [ i ]
            i += 1
        while ( i < len ( str ) and ord ( str [ i ] ) >= ord ( '1' ) and ord ( str [ i ] ) <= ord ( '9' ) ) :
            freq = freq * 10 + ord ( str [ i ] ) - ord ( '0' )
            i += 1
        for j in range ( 1, freq + 1, 1 ) :
            expand += temp
    if ( freq == 0 ) :
        expand += temp
    return expand [ k - 1 ]
